scripts/build-products.js

```diff
@@ -58,21 +58,35 @@ const createLogoDict = async logosDir => {
   return map;
 };
 
-const fetchProducts = async cover => {
+const fetchProducts = async (cover, provider) => {
   const eventFilter = cover.filters.ProductSet();
   const events = await cover.queryFilter(eventFilter);
 
   const logos = await createLogoDict(path.join(__dirname, '../src/logos'));
 
-  const ipfsHashes = events
+  const sortedEvents = events
     // sort ascending by blockNumber to get the latest ipfs hash
-    // (ascending rather than descending due to the reduce)
-    .sort((a, b) => a.blockNumber - b.blockNumber)
-    .reduce((acc, event) => {
+    // (ascending rather than descending due to the reduce into productMetadata object below)
+    .sort((a, b) => a.blockNumber - b.blockNumber);
+
+  const productMetadata = {};
+
+  await Promise.all(
+    sortedEvents.map(async event => {
       const id = event.args.id.toNumber();
       const ipfsHash = event.args.ipfsMetadata;
-      return { ...acc, [id]: { id, ipfsHash } };
-    }, {});
+      productMetadata[id] = { id, ipfsHash };
+
+      const block = await provider.getBlock(event.blockNumber);
+      const blockDate = new Date(block.timestamp * 1000);
```

Comment:
🤔 Good point! We actually only need it for comparison in sorting, so we can just use the timestamps directly. 
I converted them to Date because I was thinking of easier readability, but we won't display the dates anywhere atm and there's no other use case I can think of atm to have them as dates.
Thx for the suggestion and spotting this 🙌